parsing/lexer.mli

```diff
@@ -24,6 +24,7 @@ type error =
   | Unterminated_string_in_comment of Location.t * Location.t
   | Keyword_as_label of string
   | Literal_overflow of string
+  | Literal_malformed of string
```

Comment:
`Literal_malformed` does not sound like proper english to me. It should be `Malformed_literal` -- or maybe the more neutral `Invalid_literal`.